xbmc/addons/kodi-addon-dev-kit/include/kodi/Network.h

```diff
@@ -0,0 +1,189 @@
+#pragma once
+/*
+ *      Copyright (C) 2005-2017 Team Kodi
+ *      http://kodi.tv
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with KODI; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "AddonBase.h"
+
+/*
+ * For interface between add-on and kodi.
+ *
+ * This structure defines the addresses of functions stored inside Kodi which
+ * are then available for the add-on to call
+ *
+ * All function pointers there are used by the C++ interface functions below.
+ * You find the set of them on xbmc/addons/interfaces/General.cpp
+ *
+ * Note: For add-on development itself this is not needed
+ */
+typedef struct AddonToKodiFuncTable_kodi_network
+{
+  bool (*wake_on_lan)(void* kodiBase, const char *mac);
+  char* (*get_ip_address)(void* kodiBase);
+  char* (*dns_lookup)(void* kodiBase, const char* url, bool* ret);
+  char* (*url_encode)(void* kodiBase, const char* url);
+} AddonToKodiFuncTable_kodi_network;
+
+//==============================================================================
+///
+/// \defgroup cpp_kodi_network  Interface - kodi::network
+/// \ingroup cpp
+/// @brief **Network functions**
+///
+/// The network module offers functions that allow you to control it.
+///
+/// These are pure static functions them no other initialization need.
```

Comment:
Have removed the string complete, this description here becomes improved during completion C++ addon documentation.